signer/src/api/info.rs

```diff
@@ -0,0 +1,505 @@
+//! Handler for the `/info` endpoint.
+
+use axum::{extract::State, response::IntoResponse, Json};
+use serde::Serialize;
+
+use crate::{
+    bitcoin::BitcoinInteract,
+    config::Settings,
+    context::Context,
+    stacks::api::StacksInteract,
+    storage::{
+        model::{BitcoinBlockHash, StacksBlockHash},
+        DbRead,
+    },
+};
+
+use super::ApiState;
+
+#[derive(Debug, Serialize)]
+pub struct InfoResponse {
+    pub bitcoin: BitcoinInfo,
+    pub stacks: StacksInfo,
+    pub dkg: DkgInfo,
+    pub build_info: BuildInfo,
+    pub timestamp: String,
+}
+
+#[derive(Debug, Serialize)]
+pub struct BuildInfo {
+    pub rust_version: String,
+    pub git_revision: String,
+    pub target_arch: String,
+    pub target_env_abi: Option<String>,
+}
+
+#[derive(Debug, Serialize)]
+pub struct BitcoinInfo {
+    pub signer_tip: Option<ChainTipInfo<BitcoinBlockHash>>,
+    pub node_tip: Option<ChainTipInfo<BitcoinBlockHash>>,
+    pub node_chain: Option<String>,
+    pub node_version: Option<usize>,
+    pub node_subversion: Option<String>,
+}
+
+#[derive(Debug, Serialize)]
+pub struct StacksInfo {
+    pub signer_tip: Option<ChainTipInfo<StacksBlockHash>>,
+    pub node_tip: Option<ChainTipInfo<StacksBlockHash>>,
+    pub node_bitcoin_block_height: Option<u64>,
+    pub node_version: Option<String>,
+}
+
+#[derive(Debug, Serialize)]
+pub struct ChainTipInfo<T> {
+    pub block_hash: T,
+    pub block_height: u64,
+}
+
+#[derive(Debug, Serialize)]
+pub struct DkgInfo {
+    pub rounds: u32,
+    pub target_rounds: u32,
+    pub current_aggregate_key: Option<String>,
+    pub min_bitcoin_block_height: Option<u64>,
+}
+
+impl Default for InfoResponse {
+    fn default() -> Self {
+        // This is a build-time constant and varies per build environment,
+        // so this lint is not applicable.
+        #[allow(clippy::const_is_empty)]
+        let target_env_abi = if crate::TARGET_ENV_ABI.is_empty() {
+            None
+        } else {
+            Some(crate::TARGET_ENV_ABI.to_string())
+        };
+
+        Self {
+            bitcoin: BitcoinInfo {
```

Comment:
nit: we could derive default for `BitcoinInfo` and get the all `None` for free; also for `StacksInfo`

Reply:
[4b13bb1](https://github.com/stacks-network/sbtc/pull/1251/commits/4b13bb1ff6b758507cff56fab35497308e98ab94)